### DISCUSSION TITLE
Monospaced font, Fontfamily

### DISCUSSION THREAD

**Discussion by hue-ctrl** [TriliumNext/Trilium#1912] [Q&A]
I am still searching for a good way to "annotate" and "mark / highlight" (*not* syntax-highlight) code / program-output between normal text. For now, i paste the output to an note with [a special #cssClass](https://github.com/zadam/trilium/discussions/1780), but that makes it impossible to have "normal" text paragraphs in the same note.

I realized that it would be enough to be able to set multiple different font-families, especially a mono-spaced one, in the same note. This way, program-output could be set to monospace, and still be colored.

@zadam CKEditor seem to provide both, [font-familiy support](https://ckeditor.com/docs/ckeditor5/latest/features/font.html) and [support for marking text](https://ckeditor.com/docs/ckeditor5/latest/features/highlight.html). Is this something which could be easily activated for trilium?

**Comment by chee** [comment]
@hue-ctrl Have you tried codeblocks?

You can create one by clicking the paragraph icon on the left and selecting 'Insert code block' or by typing three backticks on an empty line.
![image](https://user-images.githubusercontent.com/178266/116865581-e4ceab80-abd7-11eb-846c-713afe929026.png)

**Reply by hue-ctrl** [reply]
sure, but you can't change the background-color of text in a code-block. You can't "mark" code.

**Reply by hue-ctrl** [reply]
Here is an example of what i mean: In the following screenshot i would like to highlight some information in the nmap output. I would be nice if i could do this in a code-block, but that is not possible.

![mark](https://user-images.githubusercontent.com/67381965/116868080-a9df6e80-abfd-11eb-9e11-83d6efc7c417.png)

**Reply by chee** [reply]
ohhh! sorry, i misunderstood! yes! it would be nice to be able to choose the font for a section of the note.

**Comment by hue-ctrl** [comment]
@zadam sorry to bother you again, but could you kindly state if this is possible with https://github.com/zadam/trilium-ckeditor5? If there are no obvious reasons for this not to work, i will try to build a custom trilium-ckeditor5 but maybe there is a reason you did not included it from the start?

**Reply by zadam** [reply]
> if this is possible

If you mean including font plugin into the CKEditor build, then yes, that should be possible with a custom build.
